app/assets/javascripts/admin/addon/lib/color-scheme-manager.js

```diff
@@ -0,0 +1,168 @@
+import { ajax } from "discourse/lib/ajax";
+
+/**
+ * determine if a scheme is the built-in default
+ * @param {Object} scheme
+ * @returns {boolean} true if built-in default scheme
+ */
+function isBuiltInDefault(scheme) {
+  return !scheme || scheme?.is_builtin_default;
+}
+
+/**
+ * get the scheme ID for API calls
+ * @param {Object} scheme
+ * @returns {number|null}
+ */
+function getSchemeId(scheme) {
+  return isBuiltInDefault(scheme) ? null : scheme?.id;
+}
+
+/**
+ * apply color scheme by updating stylesheet links
+ *
+ * @param {Object} scheme - color scheme to apply
+ * @param {Object} options
+ * @param {boolean} options.replace - replace existing tags? (default: false)
+ * @param {boolean} options.save - save changes to the server? (default: false)
+ * @returns {Promise}
+ */
+
+export async function applyColorScheme(scheme, options = {}) {
```

Comment:
Wow, that extra mile to make smooth experience over just reload is super impressive 🤩